L1Trigger/L1CaloTrigger/plugins/L1EGammaEEProducer.cc

```diff
@@ -0,0 +1,188 @@
+#include "FWCore/Framework/interface/stream/EDProducer.h"
+#include "FWCore/Framework/interface/ESHandle.h"
+#include "FWCore/Framework/interface/Event.h"
+#include "FWCore/Framework/interface/MakerMacros.h"
+
+#include "DataFormats/L1THGCal/interface/HGCalMulticluster.h"
+#include "FWCore/ParameterSet/interface/ParameterSet.h"
+#include "DataFormats/L1Trigger/interface/EGamma.h"
+#include "L1Trigger/L1CaloTrigger/interface/L1EGammaEECalibrator.h"
+#include "DataFormats/Math/interface/deltaPhi.h"
+
+// we sort the clusters in pt
+bool compare_cluster_pt(const l1t::HGCalMulticluster *cl1, const l1t::HGCalMulticluster *cl2);
+
+bool compare_cluster_pt(const l1t::HGCalMulticluster *cl1, const l1t::HGCalMulticluster *cl2) {
+  return cl1->pt() > cl2->pt();
+}
+
+int etaBin(const l1t::HGCalMulticluster *cl) {
+  float eta_min = 1.;
+  float eta_max = 4.;
+  unsigned n_eta_bins = 150;
+  int eta_bin = floor((std::abs(cl->eta()) - eta_min) / ((eta_max - eta_min) / n_eta_bins));
+  if (cl->eta() < 0)
+    return -1 * eta_bin;  // bin 0 doesn't exist
+  return eta_bin;
+}
+
+int get_phi_bin(const l1t::HGCalMulticluster *cl) {
+  float phi_min = -M_PI;
+  float phi_max = M_PI;
+  unsigned n_phi_bins = 63;
+  return floor(std::abs(reco::deltaPhi(cl->phi(), phi_min)) / ((phi_max - phi_min) / n_phi_bins));
+}
+
+pair<int, int> get_eta_phi_bin(const l1t::HGCalMulticluster *cl) { return std::make_pair(etaBin(cl), get_phi_bin(cl)); }
+
+class L1EGammaEEProducer : public edm::stream::EDProducer<> {
+public:
+  explicit L1EGammaEEProducer(const edm::ParameterSet &);
+
+private:
+  void produce(edm::Event &, const edm::EventSetup &) override;
+
+  edm::EDGetToken multiclusters_token_;
+  L1EGammaEECalibrator calibrator_;
+};
+
+L1EGammaEEProducer::L1EGammaEEProducer(const edm::ParameterSet &iConfig)
+    : multiclusters_token_(
+          consumes<l1t::HGCalMulticlusterBxCollection>(iConfig.getParameter<edm::InputTag>("Multiclusters"))),
+      calibrator_(iConfig.getParameter<edm::ParameterSet>("calibrationConfig")) {
+  produces<BXVector<l1t::EGamma>>("L1EGammaCollectionBXVWithCuts");
+}
+
+void L1EGammaEEProducer::produce(edm::Event &iEvent, const edm::EventSetup &iSetup) {
+  float minEt_ = 0;
+
+  std::unique_ptr<BXVector<l1t::EGamma>> l1EgammaBxCollection(new l1t::EGammaBxCollection);
+
+  // retrieve clusters 3D
+  edm::Handle<l1t::HGCalMulticlusterBxCollection> multiclusters_h;
+  iEvent.getByToken(multiclusters_token_, multiclusters_h);
+  const l1t::HGCalMulticlusterBxCollection &multiclusters = *multiclusters_h;
+
+  std::vector<const l1t::HGCalMulticluster *> selected_multiclusters;
+  std::map<std::pair<int, int>, std::vector<const l1t::HGCalMulticluster *>> etaphi_bins;
+
+  // here we loop on the TPGs
+  for (auto cl3d = multiclusters.begin(0); cl3d != multiclusters.end(0); cl3d++) {
+    // std::cout << "-- CL3D is EG: " <<  cl3d->hwQual() << "   "<< cl3d->eta() <<"   "<< cl3d->pt()<<std::endl;
```

Comment:
lets leave this commented code as is, useful for checks and validation.
Will clean up in future PRs.

Reply:
if there is a reason to keep commented-out code, that reason should be indicated clearly (in a separate comment). In many cases, I think the commented-out code in this PR is not needed. In general, things that are not cleaned up in the initial PR tend to persist forever.